src/test/java/com/google/api/codegen/testdata/csharp_gapic_snippets_library.baseline

```diff
@@ -0,0 +1,696 @@
+            // Snippet: AddCommentsAsync(string,IEnumerable<Comment>,CallSettings)
+            // Additional: AddCommentsAsync(string,IEnumerable<Comment>,CancellationToken)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            string formattedName = LibraryServiceClient.FormatBookName("[SHELF]", "[BOOK]");
+            IEnumerable<Comment> comments = new []
+            {
+                new Comment
+                {
+                    Comment = ByteString.CopyFromUtf8(""),
+                    Stage = Stage.UNSET,
+                },
+            };
+            // Make the request
+            await libraryServiceClient.AddCommentsAsync(formattedName, comments);
+            // End snippet
+        }
+
+        public void AddComments()
+        {
+            // Snippet: AddComments(string,IEnumerable<Comment>,CallSettings)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            string formattedName = LibraryServiceClient.FormatBookName("[SHELF]", "[BOOK]");
+            IEnumerable<Comment> comments = new []
+            {
+                new Comment
+                {
+                    Comment = ByteString.CopyFromUtf8(""),
+                    Stage = Stage.UNSET,
+                },
+            };
+            // Make the request
+            libraryServiceClient.AddComments(formattedName, comments);
+            // End snippet
+        }
+
+        public async Task GetBookFromArchiveAsync()
+        {
+            // Snippet: GetBookFromArchiveAsync(string,CallSettings)
+            // Additional: GetBookFromArchiveAsync(string,CancellationToken)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            string formattedName = LibraryServiceClient.FormatArchivedBookName("[ARCHIVE_PATH]", "[BOOK]");
+            // Make the request
+            Book response = await libraryServiceClient.GetBookFromArchiveAsync(formattedName);
+            // End snippet
+        }
+
+        public void GetBookFromArchive()
+        {
+            // Snippet: GetBookFromArchive(string,CallSettings)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            string formattedName = LibraryServiceClient.FormatArchivedBookName("[ARCHIVE_PATH]", "[BOOK]");
+            // Make the request
+            Book response = libraryServiceClient.GetBookFromArchive(formattedName);
+            // End snippet
+        }
+
+        public async Task UpdateBookIndexAsync()
+        {
+            // Snippet: UpdateBookIndexAsync(string,string,IDictionary<string, string>,CallSettings)
```

Comment:
Hmm. I've no idea whether the snippet generator works with generics at the moment. We'll have to see :)